### PR TITLE
Ensure that some instances of ExecutorService are properly shutdown

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

**Comment by geoand** [comment]
The first commit is actually wrong, I'll fix it later

**Review by gastaldi — APPROVED** [review]
LGTM

File: extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/S2iProcessor.java [hunk 1]
```diff
@@ -372,19 +373,33 @@ private static void s2iBuild(OpenShiftClient client, BuildConfig buildConfig, Fi
...
         });
+        try {
+            latch.await();
+        } catch (InterruptedException e) {
+            LOG.debug("Error waiting for build to complete.", e);
+        } finally {
+            try {
+                watch.close();
+            } catch (IOException e) {
+                LOG.debug("Error closing log reader.");
```
Comment (gastaldi) [inline extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/S2iProcessor.java:394]:
```suggestion
                LOG.debug("Error closing log reader.",e);
```

File: extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/S2iProcessor.java [hunk 2]
```diff
@@ -372,19 +373,33 @@ private static void s2iBuild(OpenShiftClient client, BuildConfig buildConfig, Fi
...
+            }
+            try {
+                executor.shutdown();
+            } catch (Exception e) {
+                LOG.debug("Error shutting down executor");
```
Comment (gastaldi) [inline extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/S2iProcessor.java:399]:
```suggestion
                LOG.debug("Error shutting down executor", e);
```

**Review by gastaldi — COMMENTED** [review]
Added some minor suggestions

**Comment by geoand** [comment]
I think `container-image-openshift` is not really used anymore (`container-image-openshift` being the way to go), which is why we aren't seeing problems.

As for the other commit, it shouldn't cause problems as the JVM terminates anyway.

**Pull request merged**